kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/manager/ResourceManager.java

```diff
@@ -0,0 +1,324 @@
+
+import io.kroxylicious.systemtests.Constants;
+import io.kroxylicious.systemtests.enums.ConditionStatus;
+import io.kroxylicious.systemtests.resources.Resource;
+import io.kroxylicious.systemtests.resources.ResourceCondition;
+import io.kroxylicious.systemtests.resources.ResourceOperation;
+import io.kroxylicious.systemtests.resources.ResourceType;
+import io.kroxylicious.systemtests.resources.kroxylicious.KroxyConfigResource;
+import io.kroxylicious.systemtests.resources.kroxylicious.KroxyDeploymentResource;
+import io.kroxylicious.systemtests.resources.kroxylicious.KroxyServiceResource;
+import io.kroxylicious.systemtests.resources.strimzi.KafkaNodePoolResource;
+import io.kroxylicious.systemtests.resources.strimzi.KafkaResource;
+import io.kroxylicious.systemtests.resources.strimzi.KafkaTopicResource;
+import io.kroxylicious.systemtests.resources.strimzi.KafkaUserResource;
+import io.kroxylicious.systemtests.utils.TestUtils;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * The type Resource manager.
+ */
+public class ResourceManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourceManager.class);
+    /**
+     * The constant STORED_RESOURCES.
+     */
+    public static final Map<String, Stack<Resource>> STORED_RESOURCES = new LinkedHashMap<>();
+    private static ResourceManager instance;
+
+    /**
+     * Gets instance.
+     *
+     * @return the instance
+     */
+    public static synchronized ResourceManager getInstance() {
+        if (instance == null) {
+            instance = new ResourceManager();
+        }
+        return instance;
+    }
+
+    private final ResourceType<?>[] resourceTypes = new ResourceType[]{
+            new KafkaResource(),
+            new KafkaTopicResource(),
+            new KafkaUserResource(),
+            new KafkaNodePoolResource(),
+            new KroxyServiceResource(),
+            new KroxyConfigResource(),
+            new KroxyDeploymentResource()
+    };
+
+    /**
+     * Create resource without wait.
+     *
+     * @param <T>  the type parameter
+     * @param testInfo the test info
+     * @param resources the resources
+     */
+    @SafeVarargs
+    public final <T extends HasMetadata> void createResourceWithoutWait(TestInfo testInfo, T... resources) {
+        createResource(testInfo, false, resources);
+    }
+
+    /**
+     * Create resource with wait.
+     *
+     * @param <T>  the type parameter
+     * @param testInfo the test info
+     * @param resources the resources
+     */
+    @SafeVarargs
+    public final <T extends HasMetadata> void createResourceWithWait(TestInfo testInfo, T... resources) {
+        createResource(testInfo, true, resources);
+    }
+
+    @SafeVarargs
+    private final <T extends HasMetadata> void createResource(TestInfo testInfo, boolean waitReady, T... resources) {
+        for (T resource : resources) {
+            ResourceType<T> type = findResourceType(resource);
```

Comment:
I think if we adopt a model where every system test gets it own ephemeral namespace, then we can do away with tracking what's been added to the namespace.  At the end of the test, we can just blow away the namespace.  The operators and kubernetes should handle the cleanup (if they don't we have a bug).

Reply:
done